query/src/lib.rs

```diff
@@ -129,6 +129,15 @@ impl PredicateFn {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Direction {
+    Ascending,
+    Descending,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Order(pub Direction, pub Variable);   // Future: Element instead of Variable?
```

Comment:
Had to look up `Element`, but that's interesting -- ordering by an aggregate.  Is there a way to bind an aggregate to a named variable, so that you don't have to repeat the aggregate in the ordering clause?